### PR TITLE
[SYCL][ESIMD] Add test to validate tanh function fix

### PR DESCRIPTION
**Pull request by fineg74**

### CODE REVIEW

**Comment by fineg74** [comment]
Complementary compiler PR https://github.com/intel/llvm/pull/7265

File: SYCL/ESIMD/regression/tanh_fix_test.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// TODO online compiler check fails for esimd_emulator
+// XFAIL: esimd_emulator
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+//==- tanh_compatibility_test.cpp - Test for tanh -==//
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/tanh_fix_test.cpp]: Should be tanh_fix_test.cpp in this line. Or it is better to rename the test to tanh_compatibility_test.cpp

File: SYCL/ESIMD/regression/tanh_fix_test.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    std::cout << "Vector test failed for " << x << std::hex << " ("
+              << *(int *)&std_result << " | " << *(int *)&out[0] << ")."
+              << std::dec << std::endl;
+    return 1;
+  }
+
+  if (std::fabs(std_result - scalar_result) > 0.000001f) {
+    std::cout << "Scalar test failed for " << x << "." << std::endl;
+    return 1;
+  }
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/tanh_fix_test.cpp]: Would you please verify all SIMD=16 elements of the VECTOR result 'out' this way?
IMO, after checking abs(std_result - scalar_result) > 0x000001, the other check for SCALAR result should check that SCALAR result is identical to VECTOR.

**Review by v-klochkov — COMMENTED** [review]
I have 3 comments. Thank you.

File: SYCL/ESIMD/regression/tanh_fix_test.cpp (hunk 2 above)
Comment (fineg74) [reply SYCL/ESIMD/regression/tanh_fix_test.cpp]: Done

File: SYCL/ESIMD/regression/tanh_fix_test.cpp [hunk 3]
```diff
@@ -76,30 +76,32 @@ int test_tanh(float x) {
...
-
   if (std::fabs(std_result - scalar_result1) > 0.000001f) {
     std::cout << "Scalar test failed for cody waite implementation for " << x
               << "." << std::endl;
     return 1;
   }
 
+  for (int i = 0; i < SIMD; ++i) {
+    if (scalar_result != out[i] &&
+        !(std::isnan(scalar_result) && std::isnan(out[i]))) {
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/tanh_fix_test.cpp:92]: What if only one of numbers is 'isnan'?

**Review by v-klochkov — APPROVED** [review]

**Comment by v-klochkov** [comment]
@fineg74 - The new LIT test failed on Windows with the fixed compiler. Please see Jenkins/llvm-test-suite:
```
[2022-11-04T17:49:58.784Z] Scalar test failed for -43.9009.
[2022-11-04T17:49:58.784Z] Scalar test failed for -43.8009.
[2022-11-04T17:49:58.784Z] Scalar test failed for -43.7009.
[2022-11-04T17:49:58.784Z] Scalar test failed for 43.699.
[2022-11-04T17:49:58.784Z] Scalar test failed for 43.799.
[2022-11-04T17:49:58.784Z] Scalar test failed for 43.899.
[2022-11-04T17:49:58.784Z] Scalar test failed for 43.999.
```

**Comment by fineg74** [comment]
> @fineg74 - The new LIT test failed on Windows with the fixed compiler. Please see Jenkins/llvm-test-suite:
> 
> ```
> [2022-11-04T17:49:58.784Z] Scalar test failed for -43.9009.
> [2022-11-04T17:49:58.784Z] Scalar test failed for -43.8009.
> [2022-11-04T17:49:58.784Z] Scalar test failed for -43.7009.
> [2022-11-04T17:49:58.784Z] Scalar test failed for 43.699.
> [2022-11-04T17:49:58.784Z] Scalar test failed for 43.799.
> [2022-11-04T17:49:58.784Z] Scalar test failed for 43.899.
> [2022-11-04T17:49:58.784Z] Scalar test failed for 43.999.
> ```

Fixed.
The only failure I see on Windows is 
kernel-bundle-merge-options-env.cpp

**Comment by fineg74** [comment]
> @fineg74 - The new LIT test failed on Windows with the fixed compiler. Please see Jenkins/llvm-test-suite:
> 
> ```
> [2022-11-04T17:49:58.784Z] Scalar test failed for -43.9009.
> [2022-11-04T17:49:58.784Z] Scalar test failed for -43.8009.
> [2022-11-04T17:49:58.784Z] Scalar test failed for -43.7009.
> [2022-11-04T17:49:58.784Z] Scalar test failed for 43.699.
> [2022-11-04T17:49:58.784Z] Scalar test failed for 43.799.
> [2022-11-04T17:49:58.784Z] Scalar test failed for 43.899.
> [2022-11-04T17:49:58.784Z] Scalar test failed for 43.999.
> ```

Fixed.
The only failing test on Windows at the moment is KernelAndProgram/kernel-bundle-merge-options-env.cpp

**Comment by steffenlarsen** [comment]
@fineg74 - The added test is passing unexpectedly on the ESIMD emulator. See https://github.com/intel/llvm/actions/runs/3427185876/jobs/5710264219.

**Pull request merged**